src/ape/utils/validators.py

```diff
@@ -0,0 +1,49 @@
+"""Base non-pydantic validator utils"""
+import re
+from warnings import warn
+
+from eth_utils import is_hex
+
+from ape.exceptions import AccountsError, AliasAlreadyInUseError
+from ape.utils.basemodel import ManagerAccessMixin
+
+MIN_PASSPHRASE_LENGTH = 6
+
+
+def _has_num(val: str):
+    return re.search(r"\d{1}", val) is not None
+
+
+def _has_special(val: str):
+    return re.search(r"[\!@#$%\^&\*\(\)]{1}", val) is not None
+
+
+def _validate_account_alias(alias: str) -> str:
+    """Validate an account alias"""
+    if alias in ManagerAccessMixin.account_manager.aliases:
+        raise AliasAlreadyInUseError(alias)
+    elif not isinstance(alias, str):
+        raise AccountsError(f"Alias must be a str, not '{type(alias)}'.")
+    elif is_hex(alias) and len(alias) >= 42:
+        # Prevents private keys from accidentally being stored in plaintext
+        # Ref: https://github.com/ApeWorX/ape/issues/1525
+        raise AccountsError("Longer aliases cannot be hex strings.")
+
+    return alias
+
+
+def _validate_account_passphrase(passphrase: str) -> str:
+    """Make sure given passphrase is valid for account encryption"""
+    if not passphrase or not isinstance(passphrase, str):
+        raise AccountsError("Account file encryption passphrase must be provided.")
+
+    if len(passphrase) < MIN_PASSPHRASE_LENGTH:
+        warn("Passphrase length is extremely short.  Consider using something longer.")
```

Comment:
Should use the warning logger so as to get access to the proper logging filtering?

Reply:
Is it something we want to be filterable?  I thought the python warning was more explicit and noticeable, kind of an in-between of an exception and just a log message.  It is used in other parts of ape, though seems to only be used for depreciation warnings.

I don't feel super strongly about warnings, but I do want to make sure the user knows they're doing something they shouldn't.  Your call.

Reply:
I'm not super opinionated on this, will let @antazoey chime in if she has a stronger opinion here.

Reply:
Historically, we have used the logger for everything, but with the pydantic refactor, I did use the deprecation warnings. Looking into best practice, i found this:

```
warnings.warn() in library code if the issue is avoidable and the client application should be modified to eliminate the warning

logging.warning() if there is nothing the client application can do about the situation, but the event should still be noted
```
(https://stackoverflow.com/questions/9595009/warnings-warn-vs-logging-warning)

This tells me that `warnings.warn` makes more sense here - as this is a user-facing warning.

We may want to audit and see where if we are using `logger.warn` anywhere we should be using `warning.warn`.

I know we are using `logger.warn` in some places that makes sense anyway, but perhaps not all (I would check CLIs first).

TIL